controllers/awscluster_controller.go

```diff
@@ -411,3 +466,101 @@ func (r *AWSClusterReconciler) requeueAWSClusterForUnpausedCluster(ctx context.C
 		}
 	}
 }
+
+func (r *AWSClusterReconciler) deleteAWSClusterFromIdentityOwnerRef(ctx context.Context, clusterScope *scope.ClusterScope) error {
+	if identityRef := clusterScope.IdentityRef(); identityRef != nil {
+		unstructuredIdentityObject, err := getUnstructuredFromObjectReference(ctx, r.Client, string(identityRef.Kind), identityRef.Name)
+		if err != nil {
+			return err
+		}
+		ownerRefToRemove := metav1.OwnerReference{
+			APIVersion: infrav1.GroupVersion.String(),
+			Kind:       awsClusterKind,
+			Name:       clusterScope.InfraClusterName(),
+			UID:        clusterScope.InfraCluster().GetUID(),
+		}
+		ownerReferences := unstructuredIdentityObject.GetOwnerReferences()
+		objectFound := false
+		if util.HasOwnerRef(ownerReferences, ownerRefToRemove) {
+			ownerReferences = util.RemoveOwnerRef(ownerReferences, ownerRefToRemove)
+			unstructuredIdentityObject.SetOwnerReferences(ownerReferences)
+			if len(ownerReferences) == 0 {
+				objectFound = removeFinalizerFromIdentity(identityRef, unstructuredIdentityObject)
+			}
+			err = r.Client.Update(ctx, unstructuredIdentityObject)
+			if err != nil {
+				return err
+			}
+		}
+		if !objectFound {
+			return errors.Errorf(notFoundErr)
+		}
+	}
+	return nil
+}
+
+func (r *AWSClusterReconciler) addAWSClusterAsOwnerForIdentityRef(ctx context.Context, clusterScope *scope.ClusterScope) error {
+	if identityRef := clusterScope.IdentityRef(); identityRef != nil {
+		unstructuredIdentityObject, err := getUnstructuredFromObjectReference(ctx, r.Client, string(identityRef.Kind), identityRef.Name)
+		if err != nil {
+			return err
+		}
+		ownerReferences := unstructuredIdentityObject.GetOwnerReferences()
+		ownerReferences = util.EnsureOwnerRef(ownerReferences, metav1.OwnerReference{
+			APIVersion:         infrav1.GroupVersion.String(),
+			Kind:               awsClusterKind,
+			Name:               clusterScope.InfraClusterName(),
+			UID:                clusterScope.InfraCluster().GetUID(),
+			BlockOwnerDeletion: aws.Bool(true),
```

Comment:
We should not block the deletion of AWSCluster when the identity is not deleted.
The definition of BlockOwnerDeletion is:
// If true, AND if the owner has the "foregroundDeletion" finalizer, then
	// the owner cannot be deleted from the key-value store until this
	// reference is removed.

Reply:
this is the add ownerRef case, so when we add the ownerRef shall we not set `BlockOwnerDeletion` to true

Reply:
so in what case we should set this? or is it not required to set it in our case?

Reply:
For example MachineDeployment is owner of MachineSet and should not be deleted before the MachineSet.

For our case, deletion of the Cluster should not be affected by the identities. Actually, there is no owner relation between identities and clusters so thinking of a better way to do this.